### PR TITLE
feat(ui): add optimized list component with focus navigation

### PR DESCRIPTION
**Pull request by aymanbagabas**
Add high-performance scrollable list component with efficient rendering:

**Core Architecture**
- Master buffer caching with lazy viewport extraction
- Dirty item tracking for partial updates
- Smart buffer manipulation for append/prepend/delete operations

**Performance Optimizations**
- Viewport height changes no longer trigger master buffer rebuilds
- In-place re-rendering when item heights unchanged
- Efficient structural operations (append/prepend/delete via buffer slicing)
- Focus navigation automatically skips non-focusable items

**Item Types**
- StringItem: Simple text with optional wrapping
- MarkdownItem: Glamour-rendered markdown with optional focus styles
- SpacerItem: Vertical spacing
- FocusableItem: Wrapper to add focus behavior to any item

**Focus Management**
- Built-in Focusable interface support
- Focus/blur state tracking with automatic item updates
- Smart navigation that skips non-focusable items (Gap, Spacer)

**Testing**
- 32 tests covering core operations, rendering, focus, scrolling
- Comprehensive test coverage for edge cases and regressions

💘 Generated with Crush

Assisted-by: Claude Sonnet 4.5 via Crush <crush@charm.land>

### CODE REVIEW

File: internal/ui/list/list.go [hunk 1]
```diff
@@ -0,0 +1,795 @@
...
+		lineIdx++
+	}
+
+	// Pad remaining lines with spaces to maintain viewport height
+	emptyLine := strings.Repeat(" ", l.width)
+	for ; lineIdx < l.height; lineIdx++ {
+		lines[lineIdx] = emptyLine
+	}
+
+	return strings.Join(lines, "\r\n")
```
Comment (kujtimiihoxha) [inline internal/ui/list/list.go:176]:
```suggestion
	return strings.Join(lines, "\n")
```
